### PR TITLE
feat: Add Dashlane secure notes support

### PR DESCRIPTION
**Pull request by XaF**
Follow-up to #2792

### CODE REVIEW

File: pkg/cmd/dashlanetemplatefuncs.go [hunk 1]
```diff
@@ -33,14 +54,13 @@ func (c *Config) dashlanePasswordTemplateFunc(filter string) any {
...
 
-	c.Dashlane.cache[filter] = data
+	c.Dashlane.cachePassword[filter] = data
 	return data
 }
 
 func (c *Config) dashlaneOutput(args ...string) ([]byte, error) {
 	name := c.Dashlane.Command
 	cmd := exec.Command(name, append(c.Dashlane.Args, args...)...)
-	cmd.Stdin = os.Stdin
```
Comment (XaF) [inline pkg/cmd/dashlanetemplatefuncs.go:43]: `dcli note` will ask the user to select a note if multiple notes are matching the filter.
Removing the access to `stdin` leads to the command line exiting on error right away when this happens.

This however does not allow to enter the master password for Dashlane if needed - that being said, those are being asked on stdout at the moment, which is not visible to the user and can only lead to confusion.

Comment (twpayne) [reply pkg/cmd/dashlanetemplatefuncs.go:43]: Thanks, makes sense.

File: pkg/cmd/testdata/scripts/dashlane.txtar [hunk 2]
```diff
@@ -64,6 +73,8 @@ IF "%*" == "password --output json filter" (
     echo.        "subdomainOnly": false
     echo.    }
     echo.]
+) ELSE IF "%*" == "note filter" (
+    echo <note>
```
[the remark this answers is not in the document]
Comment (XaF) [reply pkg/cmd/testdata/scripts/dashlane.txtar]: Updated & rebased on top of master!

**Comment by twpayne** [comment]
I can check what's causing the Windows CI step to fail later today. It's probably a syntax error in the `dcli.cmd` file in the test script.

**Comment by bradenhilton** [comment]
`<` and `>` are causing the syntax issue. They can be escaped (`^<`/`^>`) or replaced with something else.

Fixing that reveals a CRLF line-ending being added to stdout, so I elected to compare the output to a golden file instead, like in the Bitwarden tests.

```diff
diff --git a/pkg/cmd/testdata/scripts/dashlane.txtar b/pkg/cmd/testdata/scripts/dashlane.txtar
index 8ccd263b..f4dc30c4 100644
--- a/pkg/cmd/testdata/scripts/dashlane.txtar
+++ b/pkg/cmd/testdata/scripts/dashlane.txtar
@@ -1,5 +1,6 @@
 [!windows] chmod 755 bin/dcli
 [windows] unix2dos bin/dcli.cmd
+[windows] unix2dos golden/dashlane-note

 # test dashlanePassword template function
 exec chezmoi execute-template '{{ (index (dashlanePassword "filter") 0).password }}'
@@ -7,7 +8,7 @@ stdout ^<password>$

 # test dashlaneNote template function
 exec chezmoi execute-template '{{ dashlaneNote "filter" }}'
-stdout ^<note>$
+cmp stdout golden/dashlane-note

 -- bin/dcli --
 #!/bin/sh
@@ -74,8 +75,10 @@ IF "%*" == "password --output json filter" (
     echo.    }
     echo.]
 ) ELSE IF "%*" == "note filter" (
-    echo.<note>
+    echo.^<note^>
 ) ELSE (
     echo error: unknown command '$*'
     exit /b 1
 )
+-- golden/dashlane-note --
+<note>
```

```console
❯ go test ./pkg/cmd/... -run=TestScript/^dashlane$
ok      github.com/twpayne/chezmoi/v2/pkg/cmd   0.602s
```

**Comment by XaF** [comment]
Updated fixing the typo (I missed that, my bad! Thanks for pointing it out) and the change suggested by @bradenhilton above!

**Comment by twpayne** [comment]
Perfect, thank you all :)

**Pull request merged**